Src/WindowsServer/WindowsServer.Shared/Implementation/AzureRoleEnvironmentContextReader.cs

```diff
@@ -45,6 +48,27 @@ public static IAzureRoleEnvironmentContextReader Instance
                     AzureRoleEnvironmentContextReader.BaseDirectory = Path.Combine(AppDomain.CurrentDomain.SetupInformation.ApplicationBase, "bin");
                 }
 
+                if (string.IsNullOrEmpty(AzureRoleEnvironmentContextReader.AssemblyName) == true)
+                {
+                    AzureRoleEnvironmentContextReader.AssemblyName = "Microsoft.WindowsAzure.ServiceRuntime";
+                }
+
+                if (string.IsNullOrEmpty(AzureRoleEnvironmentContextReader.Culture) == true)
+                {
+                    AzureRoleEnvironmentContextReader.Culture = "neutral";
+                }
+
+                if (string.IsNullOrEmpty(AzureRoleEnvironmentContextReader.PublicKeyToken) == true)
+                {
+                    AzureRoleEnvironmentContextReader.PublicKeyToken = "31bf3856ad364e35";
+                }
+
+                if (AzureRoleEnvironmentContextReader.VersionsToAttempt == null || AzureRoleEnvironmentContextReader.VersionsToAttempt.Length == 0)
+                {
+                    // This list should be explicity updated when 3.* versions of Microsoft.WindowsAzure.ServiceRuntime are released after confirming they don't have breaking API changes.
+                    AzureRoleEnvironmentContextReader.VersionsToAttempt = new string[] { "2.7.0.0", "2.6.0.0", "2.5.0.0", "2.4.0.0", "2.3.0.0", "2.2.0.0", "2.1.0.0", "2.8.0.0", "2.9.0.0" };
```

Comment:
Here is the list I believe: https://azure.microsoft.com/en-us/downloads/archive-net-downloads/